### PR TITLE
Adding Exercises

### PR DESCRIPTION
**Pull request by MarkWignall**
Adding Exercise

### CODE REVIEW

File: chapter06/mdw/Ex01.pl [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/perl
+
```
Comment (codavex) [inline chapter06/mdw/Ex01.pl:2]: I'm gonna have to be strict and warn you again about not using strict and warnings

File: chapter06/mdw/Ex01.pl [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+use 5.010;
+
+%lastname = (
+
+  fred => 'Flintstone',
+
+  barney => 'the Dinosaur',
```
Comment (codavex) [inline chapter06/mdw/Ex01.pl:9]: rubble, surely?

File: chapter06/mdw/Ex01.pl [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+);
+
+say "Enter Firstname:";
+
+my $name = <>;
+
+chomp $name;
+
+say "Hello Mr.$lastname{$name}";
```
Comment (codavex) [inline chapter06/mdw/Ex01.pl:21]: what if the key isn't in the hash? what if the key's value is undef?

File: chapter06/mdw/Ex02.pl [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+foreach ( <> ) {
+
+  chomp $_;
+
+  if ( exists $words{$_} ) {
+
+    $words{$_} += 1;
+
+  } else {
```
Comment (codavex) [inline chapter06/mdw/Ex02.pl:15]: don't think you need this if-else clause, if it doesn't exist it should treat it as 0 and increment that

File: chapter06/mdw/Ex02.pl [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+}
+
+my @words = keys %words;
+
+@words = sort @words;
+
+foreach my $word (@words) { 
+
+  say "The word $word was used $words{$word} times";
```
Comment (codavex) [inline chapter06/mdw/Ex02.pl:29]: word!

File: chapter06/mdw/Ex03.pl [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+    $keylength = length $key;
+
+  }
+
+};
+
+$keylength += 3;
+
+my $format = "%".$keylength."s %s\n";
```
Comment (codavex) [inline chapter06/mdw/Ex03.pl:21]: nice, I forgot about defining the format outside the printf

**Comment by ardavey** [comment]
Just for the record, you absolutely must start using strict and warnings in these exercises.  Without this you are artificially limiting the benefit you're going to get from this course.

**Pull request merged**